File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    let answer = 0;
    const arr = []
    const string = [...s];

    string.forEach((char, index) => {
        arr.push(char);
        const a = arr.filter(value => value === arr[0])
        const b = arr.filter(value => value !== arr[0])
        if(a.length === b.length){
            answer++;
            arr.length = 0;
        }
    })
    if(arr.length !== 0){
        answer++;
    }
    return answer;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    cnt_x = 0
    cnt_other = 0
    x = ''
    
    for c in s:
        if cnt_x == 0:
            x = c
            cnt_x = 1
            cnt_other = 0
        else:
            if c == x:
                cnt_x += 1
            else:
                cnt_other += 1
        if cnt_x == cnt_other:
            answer+=1
            cnt_x = 0
            cnt_other = 0
            
    if cnt_x != 0:
        answer+=1
    return answer
    return answer